src/test/java/com/microsoft/sqlserver/testframework/DBResultSetMetaData.java

```diff
@@ -0,0 +1,198 @@
+// Software is furnished to do so, subject to the following conditions :
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ---------------------------------------------------------------------------------------------------------------------------------
+
+package com.microsoft.sqlserver.testframework;
+
+import static org.junit.Assert.fail;
+
+import java.sql.SQLException;
+
+import com.microsoft.sqlserver.jdbc.SQLServerResultSetMetaData;
+
+/**
+ * 
+ *
+ */
+public class DBResultSetMetaData extends AbstractParentWrapper {
+
+    /**
+     * @param parent
+     * @param internal
+     * @param name
+     */
+    DBResultSetMetaData(AbstractParentWrapper parent, Object internal, String name) {
+        super(parent, internal, name);
+        // TODO Auto-generated constructor stub
+    }
+
+    /**
+     * 
+     * @throws SQLException
+     */
+    public void verify() throws SQLException {
+        // getColumnCount
+        int columns = this.getColumnCount();
+
+        // Loop through the columns
+        for (int i = 1; i <= columns; i++) {
+            // Note: Just calling these performs the verification, in each method
+            this.getColumnName(i);
+            this.getColumnType(i);
+            this.getColumnTypeName(i);
+            this.getScale(i);
+            this.isCaseSensitive(i);
+            this.isAutoIncrement(i);
+            this.isCurrency(i);
+            this.isNullable(i);
+            this.isSigned(i);
+        }
+    }
+
+    /**
+     * 
+     * @return
+     * @throws SQLException
+     */
+    public int getColumnCount() throws SQLException {
+        int current = 0;
+        try {
+            current = ((SQLServerResultSetMetaData) product()).getColumnCount();
+        }
+        catch (SQLException e) {
+            fail(e.toString());
+        }
+        return current;
+    }
+
+    /**
+     * 
+     * @param index
+     * @return
+     * @throws SQLException
+     */
+    public String getColumnName(int index) throws SQLException {
+        String current = ((SQLServerResultSetMetaData) product()).getColumnName(index);
+        return current;
+    }
+
+    /**
+     * 
+     * @param index
+     * @return
+     * @throws SQLException
+     */
+    public int getColumnType(int index) throws SQLException {
+        int current = ((SQLServerResultSetMetaData) product()).getColumnType(index);
+        return current;
```

Comment:
can we directly return the value whenever possible rather than storing them in temp variable and then passing them? this way value will not be stored in memory while waiting for GC.